src/plugins/intel_gpu/src/graph/impls/ocl/reverse.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+#include "reverse_inst.h"
+#include "common_types.h"
+#include "primitive_base.hpp"
+#include "impls/implementation_map.hpp"
+#include "kernel_selector_helper.h"
+
+#include "reverse/reverse_kernel_selector.h"
+#include "reverse/reverse_kernel_ref.h"
+
+#include "intel_gpu/runtime/error_handler.hpp"
+
+using namespace cldnn;
+
+namespace cldnn {
+namespace ocl {
+
+struct reverse_impl : typed_primitive_impl_ocl<reverse> {
+    using parent = typed_primitive_impl_ocl<reverse>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<reverse_impl>(*this);
+    }
+
+public:
+    static primitive_impl* create(const reverse_node& arg) {
+        auto params = get_default_params<kernel_selector::reverse_params>(arg);
+        auto optional_params = get_default_optional_params<kernel_selector::reverse_optional_params>(arg.get_program());
+
```

Comment:
Looks like all local variables (except params) can be const

Reply:
done